src/zap/zapinfo.cpp

```diff
@@ -1988,12 +1994,15 @@ void * ZapInfo::getAddressOfPInvokeFixup(CORINFO_METHOD_HANDLE method,void **ppI
 
     m_pImage->m_pPreloader->AddMethodToTransitiveClosureOfInstantiations(method);
 
-    CORINFO_MODULE_HANDLE moduleHandle = m_pEECompileInfo->GetLoaderModuleForEmbeddableMethod(method);
-    if (moduleHandle == m_pImage->m_hModule 
-        && m_pImage->m_pPreloader->CanEmbedMethodHandle(method, m_currentMethodHandle))
+    if (!IsReadyToRunCompilation())
     {
-        *ppIndirection = NULL;
-        return PVOID(m_pImage->GetWrappers()->GetAddrOfPInvokeFixup(method));
+        CORINFO_MODULE_HANDLE moduleHandle = m_pEECompileInfo->GetLoaderModuleForEmbeddableMethod(method);
+        if (moduleHandle == m_pImage->m_hModule
+            && m_pImage->m_pPreloader->CanEmbedMethodHandle(method, m_currentMethodHandle))
+        {
+            *ppIndirection = NULL;
+            return PVOID(m_pImage->GetWrappers()->GetAddrOfPInvokeFixup(method));
```

Comment:
Do we need to update R2RDump to understand these signatures?

Reply:
Hmm, I'm looking at the code but it's not yet clear to me what is the exact new thing to appear in the R2R image with Fadi's change. Is that a new helper / fixup type or some previously unseen signature format flavor?

Reply:
It should be a new fixup type.

Reply:
Should be an encoded method with ENCODE_INDIRECT_PINVOKE_TARGET as the fixup type. If you point me to the R2R dump tool sources, i can make that change and test it with my pinvoke unit test, and create a separate PR for it

Reply:
Support added